### DISCUSSION TITLE
.Net 6 Upgrade Plan

### DISCUSSION THREAD

**Discussion by dev-vineet** [quartznet/quartznet#1594] [Q&A] [answered]
We have support of .net 3.1 in the quartz and EOL of .Net 3.1 is in December 3, 2022. For more infor - https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core

Microsoft has also released the .Net 6 LTS. So do we have any timeline or plan to move to .Net6?

**Comment by lahma** [comment, marked as answer]
Quartz is compatible with NET 6 and running in production using NET 6 as runtime.

**Reply by dev-vineet** [reply]
Quartz.AspNetCore is still in .Net Core 3.1. if you will use it in .net core related projects, .Net 3.1 runtime lib is used.

![image](https://user-images.githubusercontent.com/61535879/165740321-4c97c8c5-39c6-4001-90a8-87759f5dd141.png)

**Reply by lahma** [reply]
I think it should redirect just fine when running on NET 6. For reduced dependencies I think you could just use Quartz.Extensions.Hosting?

**Reply by dev-vineet** [reply]
If we will not update the project, then it will try to resolve with netcore3.1 framework. 
Yes, we can use Quartz.Extensions.Hosting directly.

Should i create a issue for it and just push the change to update the project to .Net6 ?

**Reply by lahma** [reply]
If you could create a PR against 3x branch to add net60 to target in addition to existing ones that would be great, too soon to drop 3.1.

**Reply by dev-vineet** [reply]
Thanks.
